test/hooks.test.js

```diff
@@ -3301,3 +3301,31 @@ test('onTimeout should be triggered and socket _meta is set', t => {
     })
   })
 })
+
+test('registering invalid hooks should throw an error', async t => {
```

Comment:
Could you add a test where an `onRoute` hook adds an `undefined` hook?

Here is an example code: https://www.fastify.io/docs/latest/Guides/Plugins-Guide/#hooks

Reply:
Sure, added the test in my latest commit.